paper: Initial draft SDHPC paper

jprorama opened the pull request:
Thank you for submitting to the GOOD Conference Proceedings!

Please name your PR with `<type>: <title>`, where `<type>` is either `Paper`, `Slides`, or `Poster` and `<title>` is the title of your submission. An editor will then add the `draft` label to your PR and GitHub Actions will run automated checks and create a preview build.

See the https://good-proceedings.curve.space/author-instructions for full submission instructions.

jprorama committed:
Initial draft SDHFC paper

rowanc1 commented on papers/jpr-software-defined-hpc/myst.yml:
```diff
@@ -0,0 +1,15 @@
+# See docs at: https://mystmd.org/guide/frontmatter
+version: 1
```
Can you please add this to get the preview working?

```suggestion
version: 1
extends: ../papers.yml
```

rowanc1 commented on papers/jpr-software-defined-hpc/myst.yml:
```diff
@@ -0,0 +1,15 @@
+# See docs at: https://mystmd.org/guide/frontmatter
+version: 1
+project:
+  id: 5b16e152-5730-4acd-8343-92927e2e956c
```
```suggestion
  id: good-2025-jpr-software-defined-hpc
```

rowanc1 commented on papers/jpr-software-defined-hpc/01-paper.md:
```diff
@@ -0,0 +1,172 @@
+exports:
+  - format: pdf
+    templates: volcanica
+    article_type: Report
```
```suggestion
```

These are handled in the myst extends below!

rowanc1 committed:
Update myst formatting

rowanc1 commented:
@jprorama I have done a few updates to get the PR building. You can continue to refine your draft with these changes!

davidecarlson commented:
@jprorama 

Can you please take a look at the [automated submission review](https://sites.curvenote.com/build/01989ed8-e751-7368-8fb6-49e6bd957a23#checks) and add the items that were noted as missing (author affiliation, keywords, etc.)?

Also, we noticed in the draft that it looks like the first letter in each sentence of the body of the draft is not capitalized.  Could you add capitalization as well?

Thanks!
Dave

jprorama committed:
Refined Abstract

Tightened the desicription of the infrastructure developed.
Characterize nodes as application routers instead of application proxyies
that route.
Highlight the motivating usecase of maintanence.

jprorama committed:
Rewrite Introduction

Shape introduction around the context of Open Ondemand shifting
HPC to cloud native.
This provides context for our applicaiton font ends for user routing
and the CICD workflow for its development.

Removed expsition around beowulf cluster.
Moved details on implmentation and experiment to relavent sections.

jprorama committed:
Add author affiliations and keywords to front-matter

Add missing fields to pass build checks.

jprorama committed:
Updates to related work and software defined hpc sections

Outline content the releated work section.
Rename section three as the software defined hpc.
Create RCS and CICD as subsections.
Notes for Experimental section.

jprorama committed:
Add pdf export with arxiv_two_column format

Add a familiar paper format to guide content developement
and ensure layout remains compliant with expectations.

jprorama committed:
Add RCS Architecture image

Add image for RCS subsection.
Fix minor wording in subsection.

jprorama committed:
Improvements to RCS description

Refine the dialog to separate the RCS architecture from the particular
implementation at our campus.

jprorama committed:
Finish initial draft of appliction router subsection

Motivate application routing to control the user experience
using familiar HTTP scenarios.
Introduce SSH routing with sshpiper in the context of providing
coherent control over the user experience regardless of protocol.

jprorama committed:
Add two subsections to CICD Pipelines and draft first subsection

Create subsections for
- Core Infrastructure Builds
- Deployment of HPC Services

Draft first subsection to describe the build of OOD and account app

jprorama committed:
Revise the CICD section intro

Revised the cicd section intro to provide a comprehensive overview of
pipelines leaving specifics of build and deploy to the sub sections.

jprorama committed:
Complete the CICD infrastructure build subsection

Revise the text and add paragraph descibing the image factory pattern.
Tune phrasing in preceding section text to support statements in the
subsection and avoid repeating descriptions.
Escape underscores to protect them from markdown interpretation.

jprorama committed:
Finish draft of the deployment subsection

Describe the deploy pipeline workflow.
Include the deployed components of http, ssh, and web applications.

Include some notes in the experiment section.

jprorama committed:
Add AB cluster testing image

This image drives the dialog for the experimental section.

jprorama committed:
Complete SDHPC use case section

Renamed section to highlight SDHPC use for data and
cluster migration.
Describe the use case that drove the developement of the
SDHPC approach.
Provide specfics to the the migraiton workflow, hosting
approach and use of the CICD pipelines.

Remove notes from start of section.
Move notes to section three.
Add AI Infrastructure keyword.

jprorama committed:
Complete draft of conclusion

Provide narative throughline for the topics
introduced in the paper.
Provide insights on future directions.

jprorama committed:
Write related work section

Provide a fulll narrative for relate work to tie together
the concepts we build upon in this paper.

jprorama committed:
Spell check pass

Run ispell on buffer.

jprorama committed:
Improve migration workflow description

Revise the section on blue-green model to describe the actual migration
workflow sufficiently to qualify as an experiment.
Move insights on approach as cloud-native platforms to conclusion.

jprorama committed:
Add first reference to understand MyST markup

Use bibtex in local dir and reference to prior work
on CRI_XCBC.

jprorama committed:
Add references for intro and related work sections

Improve text around references in some cases to improve
document flow and context of reference.

jprorama committed:
Revise abstract and intro

Clean up abstract and avoid parenthesis and abbreviations.
Better frame context of work in intro.
Tune intro to describe OOD architecture better.
Tune sshpiper description.

Add refrences for sshpiper and beowulf clusters.

jprorama committed:
Add section headings and cross-refs in intro

Adjust last paragraph of intro to use section headings
to refer to sections in paper.

Set section numbering to ensure all labels can be numeric.

wwarriner committed:
initial edits

jprorama committed:
Merge pull request #1 from wwarriner/ww-edit

Suggestions for intro and related work, and notes

wwarriner committed:
edits from meeting

jprorama committed:
Merge pull request #3 from wwarriner/ww-edit

Resolve conflicts in favor of ww-edit branch.
Helps to simply wordy descriptions and add
notes to for further clarification and content.

jprorama committed:
Move the environment.yml to paper dir

Attempt to fix the build error upstream related to files outside
the current dir:

Error: The strategy is set to fail when changes are made outside of a single folder (`enforce-single-folder: true`).
These files are outside of an allowed folder path:
  - environment.yml

jprorama committed:
Add main.bib to capture bibtex references

This file includes the bib references in the doc
and is built by the pdf export function.

jprorama committed:
Add William Warriner as author

Contributions to structure and review of paper.

jprorama committed:
Revise introduction to improve flow

Adjust the narrative for smoother transition
and narritive in light of the recent contributions.
Attempt to emphasize key components added to the
platform.

jprorama committed:
Address revision notes for intro to SDHPC section

Provide a transition paragraph to highlight the following
subsections.
Add references for material support of the claims.
Update bib with references.

jprorama committed:
Address notes for RCS system subsection

Revise text to fit in with section introduction.
Reduce complexity of descriptions by opting for more
consistent terminology use.
Update bib with latest references.

jprorama committed:
Condense the application router subsection

Restructure the narrative to more quickly introduce
the application routers for HTTP and SSH.
Eliminate the exposition.
Focus the account creation and states on their
utilty to providing a cloud-native experience.

jprorama commented:
@davidecarlson I'm trying to understand the failing check on the Build draft preview.  It seems to run into a error but it's not clear what is triggering the error.

Please advise on how to fix it.

jprorama committed:
Revise CICD Pipelines section intro

Tune the wording and flow of the section intro in
light of the prior section edits and goals for condensing
the following subsections.

Fix header numbering to avoid the title being numbered in
upstream integration tests.

diedpigs committed:
fix typo NFS ACCESS

jprorama committed:
Comment out subsections on build and deploy

These subsections are too detailed and repeat information
no provided in the main body of the CICD section.
Comment out the material before committing to deletion
to see if it's needed for the narrative.
Add the note about versioned builds to the build
commentary since that seems useful to know.

stevejpurves commented:
Hi @jprorama - the failing checks are related to specific citations that are missing DOIs. 

To fix this (1) to either add the DOI to each of the failing citations in your bibtex file.


or (2) if no DOIs exist for these citations, you can explicitly mark them explicitly as skipped by adding the following section to the `myst.yml` file, where the `keys` are the tags for the relevant citations in your bibtex.

```
project:
  ...
  error_rules:
    - rule: doi-exists
      severity: ignore
      keys:
        - citationTagInBibtex
```        

See the [check report](https://sites.curvenote.com/build/01999ddf-a506-7599-b5cb-b50e7bc1666f#checks) for the relevant failures.

jprorama committed:
Fix DOI checks during integration build

Make DOI optional for references since many
are URLs to online informatoion.

stevejpurves commented:
@jprorama sorry for the confusing snippet, in your case this would skip all failing checks

```
project:
  ...
  error_rules:
    - rule: doi-exists
      severity: ignore
      keys:
        - Lian2025
        - gitlab-cicd
        - openstack2025
        - CRIXCBC2024
        - Becker1999
        - Shore2021
        - Muse2023
        - carlz-at-us.ibm.com2020
        - Sedlmayer2020
        - DcbloxHDcolo
```

jprorama committed:
Merge pull request #4 from diedpigs/fix-typo

fix typo NFS ACCESS

jprorama committed:
Minor wording changes in intro and CICD subsection

Some minor word changes for clarity.

jprorama committed:
Fix citation list for to doi-exists rule ignore

Adjust the ignore rule to list the specific citation keys that
should be ignorned as receommended by @stevejpurves

jprorama committed:
Revise section 4 describing the sdhpc use case

Heavy edits to clean the narritive and remove
dead weight details.
Honed the description of the data migration
workflow.
Added the narrative for the performance testing
of sshpiper.
Reduced most discussion of the cloud hosting
to just focus on the deployed environment
and removed needless commentary.

jprorama committed:
Update main.bib references

jprorama committed:
Remove commented dead text from body

In the last few revisions commented large swaths of text
that were deemed unneccessary.
Some of the key points were incorporated into the main
text through earlier revisions of section three and four.
Remove the comments to clean the text.

jprorama committed:
Reorder sequence of summaries in conclusion

Created a better flow through the conclusion.
Drop redundent claims from earlier in the section.
Move to using infrastructure as code and add
reference to suport it during the CICD discussion.kl

jprorama committed:
Revise intro to related work transition

Add a contributions paragraph at end of introduction.
Better shape openstack position in related work to
address comments.
Remove unneeded notes, comments and text.

jprorama committed:
Revise the abstract to better describe the paper

The focus isn't exclusive to our migration use case
so make sure it highlights the construction of the
infrastructure we built.
Adjust the final paragraph of intro to better
align with abstact.
In particular make sure it mentions RCS as a
contribution.

jprorama committed:
Add refs for our CRI_XCBC fork and hpc-factory

Provides code details on the software developed
to avoid excessive description in the text.

jprorama committed:
Update author block

Include ORCID number and CRediT Roles for authors.
Update affiliation to include IT department.

jprorama committed:
Fix jpr@uab.edu ORCID

Seems the number got transposed.

jprorama committed:
Add table for ssh througput tests

Add the results for the selected tests we are reporting for
stanard clients.
Adjusted the narrative to align with the reported results,
removing commentary on latency since this was not specifically
tested.
Acknowledge need for more comprehensive testing.

jprorama committed:
Update SSH throughput test results

Update table to include the tests results gathered from the
SSH client tests in OpenStack.
https://code.rc.uab.edu/rc/load-test-for-proxy-node

Improved presentation of mean, margin of error, and
confidence interval results.

jprorama committed:
Introduce new section for SSH throughput performance analysis

Create a distinct section for the ssh performance analysis.
This focuses the performance narrative on the tested scenarios and
keeps it seperate from the discussion of the migration scenario.
Rewrote the conclusions based on the new results in the table.
Providing a full breakdown of the data in the table.

Refined the migration section discussion in like of the new section.
Moved the commentary on RCS deployment and dev support to the migration
section.
Tuned the wording in the migration section to improve flow of the
narrative.

jprorama committed:
Review abstract and introduction

Improve accuracy of abstract and narrative flow of
introduction.
Fix wording and simplify some sentances.
Rewrite the concluding paragraph of the intro that
provides an overview of the remaining sections.

jprorama committed:
Review Related Work section

Improve wording to simplify some of the narrative.

jprorama committed:
Review Software Defined HPC section

Wording improvements for clarity.
Some sentance restructures.
Add references for Packer and Terraform.
Update main.bib and doi checks in myst.yml for web refs.

jprorama committed:
Review SDHPC for Data and Cluster Migration section

Restructured the GPFS migration paragraph to better reflect
the changes that drove specific changes.
Moved the references to more specific statements.
Made the Ceph tier more explict.

Other wording changes to improve narrative flow.

jprorama committed:
Review SSH Application Router Performance section

Improved clarity of performance test interpretation for
sshpiper.
Adjusted table wording for more concise labels.
Other wording improvements for clarity.

jprorama committed:
Reviewed Conclusion section

Removed first paragraph.
Revised remaining text for concise statments and more effective
expressions of conclusions.
Overall improvement to coherency in the conclusion.

jprorama committed:
Add acknowledgments to header and spell check

Include acknowledgments as an appreciation.
Fix spelling errors.
Add orcid for Atluri
Capture changes to main.bib from reference order changes.

jprorama committed:
Fix bib typo

The case of a bib entry was inconsist with the myst exception causing
a build failure on the PR.

jprorama committed:
Fix additional typo in bib entry

Update citation in paper.

jprorama committed:
Edits to Acknowledgements

Fix wording and typo.

wwarriner committed:
intro

wwarriner committed:
related work

wwarriner committed:
sdhpc

wwarriner committed:
rcs

wwarriner committed:
app routers

wwarriner committed:
cicd pipelines

wwarriner committed:
migration

wwarriner committed:
performance

wwarriner committed:
conclusion

wwarriner committed:
alternative rcs diagram

jprorama committed:
Merge branch 'rc3_warr' into feat-paper-jpr-software-defined-hpc

Accept edit review and new rcs diagram.

Closes #5

jprorama committed:
Additional edits post review

Changed the description of the sshpiper functionalty to use
daisy-chaining which is much more explicit.
Pipelining implies multiple transactions over the same channel.
Additional minor wording changes and improved statments.

jprorama committed:
Replace table with time-based anlysis results

Address critique that the experimental results should anlyze
the direct time readings and treat throughput as a derived value.
Adjust the values in the table and descriptions associated with the
interpretation.

wwarriner committed:
update rcs arch figure

jprorama committed:
Merge pull request #6 from wwarriner/update-rcs-arch-figure

Update rcs arch figure

jprorama committed:
Add krish orcid

jprorama committed:
Update acknowledgements

Add NSF awards supporting IU engagement.

jprorama committed:
Improve RCS description

Add paragraph detailing the single computer mapping of the
RCS model and as simplification for understaning complex
systems.
Adjust the figure description to relect the image updates.
Wording improvements to improve the flow of the section as
whole.

jprorama committed:
Spell check corrections

jprorama commented:
Hi,

We are doing a final read through of the paper but feel ready to call it complete. We were curious what steps will occur to remove the draft label and merge the PR.   

Thanks for helping us understand the process.

stevejpurves commented:
@jprorama thanks - next step is for one of @davidecarlson @tkrobatsch or @luop0812 to review and approve

jprorama committed:
Review through Section3.1

Typo fixes.
Word tuning.
Hightlight change for releated work warehouse scale computing.
Edit RCS description to incorporate to-be-deleted wsc PC analogy.

wwarriner committed:
spag and clarity adjustments

jprorama committed:
Merge pull request #7 from wwarriner/ww-spag-clarity

spag and clarity adjustments

jprorama committed:
Revise the data-center-as-computer related work

Focus the reference around the warehouse scale computing model
replacing the leap to the direct PC analogy.

jprorama committed:
Revise pharasing, fix order and misc wording improvments

Revised wording for Jetstrean2 and CRI_XCBC descriptions.
Fixed sentance ordering in throughput testing.
Improved wording in a few places.

jprorama commented:
@stevejpurves Thanks for the details.  We are ready to request our merge review at this point.

jprorama commented:
Correcting my miss-directed request.   

@davidecarlson @tkrobatsch or @luop0812 we are ready for our merge review for publication.

jprorama commented:
Withdrawing paper.
